src/box/txn_limbo.c

```diff
@@ -427,26 +433,29 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
 
 	assert(!txn_limbo_is_empty(limbo));
 	struct txn_limbo_entry *e;
-	bool is_first_waiting_entry = true;
-	rlist_foreach_entry(e, &limbo->queue, in_queue) {
-		if (e == entry)
-			break;
-		if (txn_has_flag(e->txn, TXN_WAIT_ACK) &&
-		    e->txn->fiber != NULL) {
-			is_first_waiting_entry = false;
-			break;
+	if (!box_begin_timeout_new_meaning &&
```

Comment:
I didn't get, why do you need to make that conditional

Reply:
As we have decided (perhaps temporarily), if `compat.box_cfg_replication_synchro_timeout = 'old'` and `compat.box_begin_timeout_meaning = 'new'`, then we prefer `box_begin({timeout=...})` (`replication_synchro_timeout` not actually starts). And that's just what I do here.
Here we handle the rollback by `replication_synchro_timeout`. And we know that the `replication_synchro_timeout` timer was only started `if (!box_begin_timeout_new_meaning && replication_synchro_timeout_rollback_enabled)`:
```cpp
if (box_begin_timeout_new_meaning) {
	double timeout = txn->begin_time +
		txn->timeout - fiber_clock();
	rc = fiber_cond_wait_timeout(
		&limbo->wait_cond, timeout);
} else if (replication_synchro_timeout_rollback_enabled) {
	double timeout = start_time +
		replication_synchro_timeout - fiber_clock();
	rc = fiber_cond_wait_timeout(
		&limbo->wait_cond, timeout);
} else {
	rc = fiber_cond_wait(&limbo->wait_cond);
}
```